sensor/common/processsignal/enricher.go

```diff
@@ -25,10 +27,29 @@ type enricher struct {
 }
 
 type containerWrap struct {
-	processes  []*storage.ProcessIndicator
+	lock sync.RWMutex
+	// processes is an append-only doubly linked list of process indicators.
+	processes  list.List
```

Comment:
What's the motivation to switch to a `list.List` here? You call `PushBack` in the critical section, that would also work with `append`ing to a slice. You do iterate over the list below outside of a critical section. However, I see nothing in the `list.List` documentation that suggest that this is valid. I understand that because it's append-only you know that the `next` pointer for the first `n` elements will never change, but uh, that is using a lot of under-the-hood implementation knowledge (a.o.t. the fact that `MoveBefore` and the like are never called).

You can safely leave this as a slice, as long you guard operations on `cw.processes` through an `(RW)Mutex`. I was first thinking of doing a copy-on-read `getProcesses() []*storage.ProcessIndicator`, but I think you don't even need copy-on-read. Let me comment more below.